packages/jupyter-ai/src/components/chat-messages.tsx

```diff
@@ -106,24 +105,32 @@ export function ChatMessages(props: ChatMessagesProps) {
   const [timestamps, setTimestamps] = useState<Record<string, string>>({});
 
   /**
-   * Effect: update cached timestamp strings upon receiving a new message and
-   * every 5 seconds after that.
+   * Effect: update cached timestamp strings upon receiving a new message.
    */
   useEffect(() => {
     function updateTimestamps() {
       const newTimestamps: Record<string, string> = {};
```

Comment:
You will want to define `newTimestamps = { ...timestamps }`. Otherwise `(!(message.id in newTimestamps))` will always be `true`, since `newTimestamps` is an empty object.